lib/sqlalchemy/sql/compiler.py

```diff
@@ -3323,14 +3335,47 @@ def _generate_prefixes(self, stmt, prefixes, **kw):
             clause += " "
         return clause
 
-    def _render_cte_clause(self):
+    def _render_cte_clause(
+        self,
+        nesting_level=None,
+    ):
+        if not self.ctes:
+            return ""
+
+        ctes = self.ctes
+
+        if nesting_level and nesting_level > 1:
+            ctes = {
+                cte: ctes[cte]
+                for cte in ctes
+                if cte.nesting and self.level_by_ctes[cte] == nesting_level
+            }
+
+            if not ctes:
+                return ""
+
+            # Remove them from the visible CTEs
+            self.ctes = {
+                cte: self.ctes[cte]
+                for cte in self.ctes
+                if not cte.nesting and self.level_by_ctes[cte] == nesting_level
+            }
+
+            if ctes and not self.dialect.supports_nesting_cte:
```

Comment:
we tend to not do checks like this because they prohibit the construct from being usable on a database version that adds new support for a feature.  in this case MariaDB for example could support the syntax in a new version, and users should be able to use it.

as far as raising an error if the syntax isn't supported, the database does that already.   so usually we try to let the database raise for unsupported syntaxes.